src/Pages/SignUp/SignUp.js

```diff
@@ -1,7 +1,170 @@
-import React from "react";
+import React, { useState } from "react";
+import { useHistory } from "react-router-dom";
+import styled from "styled-components";
+import SignUpInput from "./SignUpInput";
 
 const SignUp = () => {
-  return <div>signup</div>;
+  let [userName, setUserName] = useState("");
+  let [userEmail, setUserEmail] = useState("");
+  let [userPw, setUserPw] = useState("");
+
+  const history = useHistory();
+  const nameCheck = userName !== "" && userName.length < 3;
+  const emailCheck = userEmail !== "" && !userEmail.includes("@");
+  const pwCheck = userPw !== "" && userPw.length < 7;
+  const check =
+    userName.length >= 2 && userEmail.includes("@") && userPw.length > 7;
+
+  const SignUpDatas = [
+    {
+      name: userName,
+      title: "회원이름",
+      type: "text",
+      validation: nameCheck,
+      inputAlert: "두 글자 이상의 이름을 입력해주세요.",
+      setName: setUserName,
+    },
+    {
+      name: userEmail,
+      title: "이메일주소",
+      type: "email",
+      validation: emailCheck,
+      inputAlert: "정확한 이메일 주소으로 입력해주세요.",
+      setName: setUserEmail,
+    },
+    {
+      name: userPw,
+      title: "비밀번호",
+      type: "password",
+      validation: pwCheck,
+      inputAlert: "8자 이상 영문 대 소문자, 숫자, 특수문자를 사용하세요.",
+      setName: setUserPw,
+    },
+  ];
+
+  const handleBtn = () => {
+    fetch("http://10.58.7.101:8000/users/sign-up", {
+      method: "POST",
+      body: JSON.stringify({
+        name: userName,
+        email: userEmail,
+        password: userPw,
+        signup_type: "flix",
+      }),
+    })
+      .then(res => res.json())
+      .then(res => {
+        if (res.message === "SUCCESS") {
+          alert(`${userName}님, 가입을 환영합니다.`);
+          history.push("/login");
+        } else {
+          alert("조건에 맞게 기입해주세요.");
+        }
+      });
+  };
+
+  return (
+    <Whole>
+      <Logo>
+        <img src="./images/logo.png" alt="logo" />
+      </Logo>
+      <SignUpContainer>
+        <Title>회원가입</Title>
+        {SignUpDatas.map((SignUpData, idx) => {
+          return <SignUpInput key={idx} data={SignUpData} />;
+        })}
+        <Btn disabled={!check} onClick={() => handleBtn()}>
```

Comment:
```suggestion
        <Btn disabled={!check} onClick={handleBtn}>
```
함수 자체를 넘겨줘도 동일한 결과일 거 같습니다

Reply:
반영했습니다! 👍🏻